### DISCUSSION TITLE
Delta apply failed - changes pane shows error

### DISCUSSION THREAD

**Discussion by cycloxslug** [opengisch/QField#2969] [Bugs]
We created a QGIS project on the desktop, synced it to the cloud, including associated shp files (looking back at the documentation, perhaps the shp file is the problem - and should have been a geopackage?)
In the field, we used QField Cloud to collect data points into that shp layer. At the end of data collection, we pushed changes. The QFieldCloud project summary showed 833 changes, but looking at the changes pane, all resulted in an error. Looking at one individual change, the map view says "Geometry not modified", however the Raw JSON appears to include the location for that point.

`{
  "new": {
    "geometry": "Point (1543261.0035321437753737 559057.87892247852869332)",
    "attributes": {
      "fid": 827,
      "Substrate": "S"
    }
  },
  "uuid": "e3ffced7-12b2-466f-b0da-f4add9b19899",
  "method": "create",
  "localPk": "827",
  "clientId": "c5be94bc-c627-49e0-8ad3-4bc5bfa8719c",
  "exportId": "9df07ffa-e617-4d39-b078-6a17308ece35",
  "sourcePk": "827",
  "localLayerId": "CaesarCreek_Substrate_GT_cc950e35_1b75_4da0_81da_34350a5ba7d6",
  "localLayerCrs": "PROJCRS[\"NAD_1983_CORS96_StatePlane_Ohio_South_FIPS_3402_Ft_US\",BASEGEOGCRS[\"NAD83(CORS96)\",DATUM[\"NAD83 (Continuously Operating Reference Station 1996)\",ELLIPSOID[\"GRS 1980\",6378137,298.257222101,LENGTHUNIT[\"metre\",1]]],PRIMEM[\"Greenwich\",0,ANGLEUNIT[\"degree\",0.0174532925199433]],ID[\"EPSG\",6783]],CONVERSION[\"NAD_1983_CORS96_StatePlane_Ohio_South_FIPS_3402_Ft_US\",METHOD[\"Lambert Conic Conformal (2SP)\",ID[\"EPSG\",9802]],PARAMETER[\"Latitude of false origin\",38,ANGLEUNIT[\"degree\",0.0174532925199433],ID[\"EPSG\",8821]],PARAMETER[\"Longitude of false origin\",-82.5,ANGLEUNIT[\"degree\",0.0174532925199433],ID[\"EPSG\",8822]],PARAMETER[\"Latitude of 1st standard parallel\",38.7333333333333,ANGLEUNIT[\"degree\",0.0174532925199433],ID[\"EPSG\",8823]],PARAMETER[\"Latitude of 2nd standard parallel\",40.0333333333333,ANGLEUNIT[\"degree\",0.0174532925199433],ID[\"EPSG\",8824]],PARAMETER[\"Easting at false origin\",1968500,LENGTHUNIT[\"US survey foot\",0.304800609601219],ID[\"EPSG\",8826]],PARAMETER[\"Northing at false origin\",0,LENGTHUNIT[\"US survey foot\",0.304800609601219],ID[\"EPSG\",8827]]],CS[Cartesian,2],AXIS[\"easting (X)\",east,ORDER[1],LENGTHUNIT[\"US survey foot\",0.304800609601219]],AXIS[\"northing (Y)\",north,ORDER[2],LENGTHUNIT[\"US survey foot\",0.304800609601219]],USAGE[SCOPE[\"Not known.\"],AREA[\"United States (USA) - Ohio - counties of Adams; Athens; Belmont; Brown; Butler; Champaign; Clark; Clermont; Clinton; Darke; Fairfield; Fayette; Franklin; Gallia; Greene; Guernsey; Hamilton; Highland; Hocking; Jackson; Lawrence; Licking; Madison; Meigs; Miami; Monroe; Montgomery; Morgan; Muskingum; Noble; Perry; Pickaway; Pike; Preble; Ross; Scioto; Vinton; Warren; Washington.\"],BBOX[38.4,-84.83,40.36,-80.7]],ID[\"ESRI\",103509]]",
  "sourceLayerId": "CaesarCreek_Substrate_GT_cc950e35_1b75_4da0_81da_34350a5ba7d6",
  "localLayerName": "CaesarCreek_Substrate_GT"
}`

The jobs pane shows Delta Apply as failed, with this information in the console:
`Download Project Directory
INFO:root:Downloading file "projects/fa2e9bc3-033a-4507-bf47-80405927bd08/files/CaesarCreek_Substrate_GT.cpg", size: 5 bytes, md5sum: "ae3b3df9970b49b6523e608759bc957d" 
INFO:root:Downloading file "projects/fa2e9bc3-033a-4507-bf47-80405927bd08/files/CaesarCreek_Substrate_GT.dbf", size: 66 bytes, md5sum: "890d500642b79ea40a61054db52a35b4" 
INFO:root:Downloading file "projects/fa2e9bc3-033a-4507-bf47-80405927bd08/files/CaesarCreek_Substrate_GT.prj", size: 544 bytes, md5sum: "b49068d11294906887e2f6e9c76086dd" 
INFO:root:Downloading file "projects/fa2e9bc3-033a-4507-bf47-80405927bd08/files/CaesarCreek_Substrate_GT.shp", size: 100 bytes, md5sum: "54ded22b0469a096beec9d05e0494090" 
INFO:root:Downloading file "projects/fa2e9bc3-033a-4507-bf47-80405927bd08/files/CaesarCreek_Substrate_GT.shx", size: 100 bytes, md5sum: "54ded22b0469a096beec9d05e0494090" 
INFO:root:Downloading file "projects/fa2e9bc3-033a-4507-bf47-80405927bd08/files/Habitat_Groundtruthing.qgs", size: 44940 bytes, md5sum: "71c1cff12458fc6f3b6905f9f438e0b4" 
INFO:root:Downloading file "projects/fa2e9bc3-033a-4507-bf47-80405927bd08/files/substrate_field.cpg", size: 5 bytes, md5sum: "ae3b3df9970b49b6523e608759bc957d" 
INFO:root:Downloading file "projects/fa2e9bc3-033a-4507-bf47-80405927bd08/files/substrate_field.dbf", size: 189 bytes, md5sum: "91dbf1b64d254107442d450d3830c4d5" 
INFO:root:Downloading file "projects/fa2e9bc3-033a-4507-bf47-80405927bd08/files/substrate_field.prj", size: 544 bytes, md5sum: "b49068d11294906887e2f6e9c76086dd" 
INFO:root:Downloading file "projects/fa2e9bc3-033a-4507-bf47-80405927bd08/files/substrate_field.shp", size: 128 bytes, md5sum: "9ce2c68afef0dab7e57ba88172787533" 
INFO:root:Downloading file "projects/fa2e9bc3-033a-4507-bf47-80405927bd08/files/substrate_field.shx", size: 108 bytes, md5sum: "2d5df7fb6477bd6aef0dc6c0498aa99b" 
Apply Deltas
INFO:QGIS_STDERR:Starting QGIS app version 32402 (exported)...
WARNING:QGIS_STDERR:QStandardPaths: XDG_RUNTIME_DIR points to non-existing path '/run/user/0', please create it with 0700 permissions.
INFO:root:Loading project file "/tmp/tmp310ej3mb/files/Habitat_Groundtruthing.qgs"...
INFO:root:Loading delta file "/io/deltafile.json"...
INFO:qfieldcloud.qgis.apply_deltas:Successfully applied delta on layer "CaesarCreek_Substrate_GT_cc950e35_1b75_4da0_81da_34350a5ba7d6"
ERROR:qfieldcloud.qgis.apply_deltas:An unknown error has been encountered while applying delta: 
INFO:QGIS_STDERR:Stopping QGIS app...`

Perhaps mistakenly, we then selected "Synchronize" from QFieldCloud, and all the points have disappeared from the QFieldCloud app and all errors remain. Manually applying changes also fails. 

2 questions:
How can we recover the data points? (it seems like their information is all present in the cloud, just somehow not translated into the shp file)
What do we need to do differently to successfully sync data?

**Comment by kartografs** [comment]
same problem here.

**Reply by suricactus** [reply]
@kartografs can you post another question with details what you did and what you see. Adding you project name would help a lot for internal investigation.

**Comment by suricactus** [comment]
Hey @cycloxslug , I have checked your issue and I will need additional time to find the root cause of your error. Just to let you know we are working on it. If you can add any additional context it would be much appreciated!

**Reply by cycloxslug** [reply]
Based on what my student has seen so far - we're pretty sure this is related to a shp file being the layer where data is being added. He set up a similar project with a gdb instead of a shp and it seems to sync without any errors.

**Comment by OBVYamaska** [comment]
I had a very similar issue a month ago that is still not resolved. I'm starting to consider my data lost...

https://github.com/opengisch/QField/discussions/2867

**Comment by hasiber** [comment]
I have the same problem with geopackage in project. I apply the changes but the error don't disappear. 
in the project overview the status is 'Resolve changes'

user: miwe, project: karte_2022
